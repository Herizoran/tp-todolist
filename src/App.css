.App {
  text-align: center;

}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
.heading{
  width: 181px;
  height: 28px;

  font-family: "Adobe Fangsong Std R", sans-serif;
  font-weight: 800;
  font-size: 25px;
  line-height: 23px;
  text-align: center;
  padding: 10px;
  margin: 10px;

  color: #000000;
}
.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
