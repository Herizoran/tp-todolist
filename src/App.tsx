import React, {useState} from 'react';
import './App.css';
import InputField from "./components/InputField";
import {Todo} from "./model";
import TodoList from "./components/TodoList";

function App() {
    const [todo, setTodo] = useState<string>("");
    const [todos, setTodos] = useState<Todo[]>([]);

    const handleSubmit = (event:React.FormEvent) => {
        event.preventDefault();
        if (todo){
            setTodos([...todos, {id:Date.now(),todo,isDone:false}])
            setTodo("");
        }
    }
    return (
    <div className="App">
      <span className="heading">TODOLIST</span>
        <InputField todo ={todo} setTodo = {setTodo} handleSubmit = {handleSubmit} />
        <TodoList todos ={todos} setTodos = {setTodos}/>
    </div>
  );
}

export default App;
