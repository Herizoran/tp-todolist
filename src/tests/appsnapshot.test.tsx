import React from "react";
import { cleanup, render } from "@testing-library/react";
import App from "../App";

afterEach(cleanup);
describe("App Snapshot", () => {
  test("TodoList component matches snapshot", () => {
    const { asFragment } = render(<App />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});