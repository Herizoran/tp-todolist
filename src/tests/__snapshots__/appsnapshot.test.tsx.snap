// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot TodoList component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <span
      class="heading"
    >
      TODOLIST
    </span>
    <form
      class="input"
    >
      <input
        class="input-box"
        data-testid="todo-input"
        placeholder="Enter Task"
        type="input"
        value=""
      />
    </form>
    <div
      class="todos-layout"
    >
      <div
        class="todos-active"
      >
        <h2>
          ACTIVE LIST
        </h2>
        <div
          class="todos"
        />
      </div>
      <div
        class="todos-done"
      >
        <h2>
          DONE
        </h2>
        <div
          class="todos"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TodoList component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <span
      class="heading"
    >
      TODOLIST
    </span>
    <form
      class="input"
    >
      <input
        class="input-box"
        data-testid="todo-input"
        placeholder="Enter Task"
        type="input"
        value=""
      />
    </form>
    <div
      class="todos-layout"
    >
      <div
        class="todos-active"
      >
        <h2>
          ACTIVE LIST
        </h2>
        <div
          class="todos"
        />
      </div>
      <div
        class="todos-done"
      >
        <h2>
          DONE
        </h2>
        <div
          class="todos"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
